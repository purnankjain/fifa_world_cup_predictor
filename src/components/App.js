import React, { Component } from 'react';
import '../styles/css/App.css';
import GroupTable from './GroupTable';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      groupAWinners : [],
      groupBWinners : [],
      semiFinal1Winner : '',
      semiFinal2Winner : '',
    };
    this.groupAWinnersUpdateCallback = this.groupAWinnersUpdateCallback.bind(this);
    this.groupBWinnersUpdateCallback = this.groupBWinnersUpdateCallback.bind(this);
    this.semiFinal1WinnerUpdateCallback = this.semiFinal1WinnerUpdateCallback.bind(this);
    this.semiFinal2WinnerUpdateCallback = this.semiFinal2WinnerUpdateCallback.bind(this);
    this.finalWinnerUpdateCallback = this.finalWinnerUpdateCallback.bind(this);
  }
  groupAWinnersUpdateCallback(groupWinners) {
    this.setState({groupAWinners: groupWinners});
  }
  groupBWinnersUpdateCallback(groupWinners) {
    this.setState({groupBWinners: groupWinners});
  }
  semiFinal1WinnerUpdateCallback(groupWinners) {
    this.setState({semiFinal1Winner: groupWinners[0]});
  }
  semiFinal2WinnerUpdateCallback(groupWinners) {
    this.setState({semiFinal2Winner: groupWinners[0]});
  }
  finalWinnerUpdateCallback(groupWinners) {
    this.setState({finalWinner: groupWinners[0]});
  }
  render() {
    return (<div>
      <GroupTable name={'Group A'} teams={['India', 'Pakistan', 'England', 'Sri Lanka']} winnerCount={2} winnerUpdateCallback={this.groupAWinnersUpdateCallback}/>
      <GroupTable name={'Group B'} teams={['Austrailia', 'New Zealand', 'West Indies', 'South Africa']} winnerCount={2} winnerUpdateCallback={this.groupBWinnersUpdateCallback}/>
      <GroupTable name={'Semi Finals 1'} teams={[this.state.groupAWinners[0], this.state.groupBWinners[1]]} winnerUpdateCallback={this.semiFinal1WinnerUpdateCallback}/>
      <GroupTable name={'Semi Finals 2'} teams={[this.state.groupAWinners[1], this.state.groupBWinners[0]]} winnerUpdateCallback={this.semiFinal2WinnerUpdateCallback}/>
      <GroupTable name={'Finals'} teams={[this.state.semiFinal1Winner, this.state.semiFinal2Winner]} winnerUpdateCallback={this.finalWinnerUpdateCallback}/>
      <GroupTable name={'Winner'} teams={[this.state.finalWinner]} winnerCount={0} />
    </div>);
  }
}

export default App;
