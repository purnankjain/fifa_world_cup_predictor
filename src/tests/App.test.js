import React from 'react';
import App from '../components/App';
import renderer from 'react-test-renderer';

describe('App Tests', () => {
  it('renders without crashing', () => {
    var rendered = renderer.create(<App />).toJSON();
  
    expect(rendered).toMatchSnapshot();
  });  
});
