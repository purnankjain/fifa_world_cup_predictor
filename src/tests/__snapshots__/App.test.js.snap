// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests renders without crashing 1`] = `
<div>
  <div>
    <h3>
      Group A
    </h3>
    <div
      className=""
      onClick={[Function]}
    >
      India
    </div>
    <div
      className=""
      onClick={[Function]}
    >
      Austr
    </div>
    <div
      className=""
      onClick={[Function]}
    >
      West
    </div>
  </div>
  <div>
    <h3>
      Group B
    </h3>
    <div
      className=""
      onClick={[Function]}
    >
      Brazil
    </div>
    <div
      className=""
      onClick={[Function]}
    >
      Arg
    </div>
    <div
      className=""
      onClick={[Function]}
    >
      Chile
    </div>
  </div>
  <div>
    <h3>
      Group C
    </h3>
    <div
      className=""
      onClick={[Function]}
    >
      USA
    </div>
    <div
      className=""
      onClick={[Function]}
    >
      Canada
    </div>
    <div
      className=""
      onClick={[Function]}
    >
      England
    </div>
  </div>
</div>
`;
